### DISCUSSION TITLE
Disabled Select options

### DISCUSSION THREAD

**Discussion by darren-glanville** [filamentphp/filament#6702] [Help]
How do you go about disabling specific options, I still want them shown and selected when the record includes them as I would like to only show active users if they have an active subscription.

Thanks

Darren

**Comment by ibrahimBougaoua** [comment]
You can use disabled function.

For example : Toggle::make('is_admin')->disabled(! auth()->user()->isAdmin())

in you'r case you can define function hasSubscription()

Select::make('patient_id')
->label('Patient')
->options(Patient::pluck('name','id')->toArray())
->disabled(! auth()->user()->hasActiveSubscription())

https://filamentphp.com/docs/2.x/forms/fields#disabling

And you can use it with hidden function :

TextInput::make('option')->hidden(! auth()->user()->hasActiveSubscription())

https://filamentphp.com/docs/2.x/forms/fields#hiding

**Reply by darren-glanville** [reply]
Thanks for the reply, it is for the individual options that I am looking for and not the who filed as per the answer.

There is vendor/filament/forms/src/Components/Concerns/CanDisableOptions.php, is there a way of using this?

**Reply by pboivin** [reply]
`->disableOptionWhen(fn ($value) => in_array($value, [ ... ]))`

**Reply by darren-glanville** [reply]
Sorry not working, this is what I have:
`Forms\Components\Select::make("user_id")
                                    ->label(__('ev-bookings.fields.member-name'))
                                    ->options(User::orderBy('name', 'ASC')->pluck('name', 'id')->toArray())
                                    ->disableOptionWhen(fn ($value) => !in_array($value, $active_users))
                                    ->searchable()
                                    ->hidden(fn (Closure $get) => $get('system'))
                                    ->required(),`

Any clues would be great?

**Reply by pboivin** [reply]
Looks ok... I'm not sure. Do you get disabled items if you try `->disableOptionWhen(fn ($value) => true)`?

**Reply by darren-glanville** [reply]
No, all the options are still active.

**Reply by pboivin** [reply]
Which version of Filament?

**Reply by darren-glanville** [reply]
Just found that if searchable() is used then the disableOptionWhen() does not work. So now if I remove searchable() is works so its a partial answer.

Would be good to be able to used searchable() with it.
